nextcord/mixins.py

```diff
@@ -65,7 +73,55 @@ async def pins(self) -> List[Message]:
             The messages that are currently pinned.
         """
 
+        warnings.warn(
+            ".pins is deprecated, use .fetch_pins instead.",
+            stacklevel=2,
+            category=FutureWarning,
+        )
+
         channel = await self._get_channel()
         state = self._state
         data = await state.http.pins_from(channel.id)
         return [state.create_message(channel=channel, data=m) for m in data]
+
+    async def fetch_pins(
+        self, limit: int = 50, before: Optional[SnowflakeTime] = None
+    ) -> AsyncIterator[MessagePin]:
+        """|asynciter|
+
+        Returns an async iterator of all messages that are currently pinned in the channel.
+
+        This requires :attr:`~nextcord.Permissions.read_message_history`, otherwise the result will be empty.
```

Comment:
According to the docs, view_channel is required too or else it'll return empty.

Reply:
I avoided putting that in as to match `history`, but I suppose that could be added too.